File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/variables/types/AsyncVariableType.java
package ch.rmy.android.http_shortcuts.variables.types;

import android.content.Context;

import org.jdeferred.Deferred;

import ch.rmy.android.http_shortcuts.realm.Controller;
import ch.rmy.android.http_shortcuts.realm.models.Variable;
import ch.rmy.android.http_shortcuts.variables.Showable;

public interface AsyncVariableType {

    Showable createDialog(Context context, Controller controller, Variable variable, Deferred<String, Void, Void> deferredValue);

    boolean hasTitle();

}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/variables/types/AsyncVariableType.kt
package ch.rmy.android.http_shortcuts.variables.types

import android.content.Context
import ch.rmy.android.http_shortcuts.data.models.Variable
import io.reactivex.Single

interface AsyncVariableType {

    fun resolveValue(context: Context, variable: Variable): Single<String>

    val hasTitle: Boolean

}
